### ISSUE TITLE
Quick Note

### ISSUE THREAD

**Issue by AlbaHoo** [AlbaHoo/cheeger.blog#23]
https://cheeger.com/general/2019/11/29/quick-notes.html 

 Quick Note

**Comment by AlbaHoo** [comment]
# android cpu is 32 or 64?
kvim:/ $ cat /pro
proc/                property_contexts
kvim:/ $ cat /proc/cpuinfo
Processor	: AArch64 Processor rev 4 (aarch64)
processor	: 0
processor	: 1
processor	: 2
processor	: 3
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer	: 0x41
CPU architecture: AArch64
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

Hardware	: Amlogic
Serial		: 210d8400bd176b7f4c2e0cf6f3885b8a
kvim:/ $ %

**Comment by AlbaHoo** [comment]
# Fragment detach vs destroy
You'll notice that when a Fragment is detached, its onPause, onStop and onDestroyView methods are called only (in that order). On the other hand, when a Fragment is removed, its onPause, onStop, onDestroyView, onDestroy and onDetach methods are called (in that order). Similarly, when attaching, the Fragment's onCreateView, onStart and onResume methods are called only; and when adding, the Fragment's onAttach, onCreate, onCreateView, onStart and onResume methods are called (in that order). – Adil Hussain

**Comment by AlbaHoo** [comment]
Updating:

Using adb shell input:

Insert text:

adb shell input text "insert%syour%stext%shere"
(obs: %s means SPACE)

..

Event codes:

adb shell input keyevent 82
(82 ---> MENU_BUTTON)

"For more keyevents codes see list below"

..

Tap X,Y position:

adb shell input tap 500 1450
To find the exact X,Y position you want to Tap go to:

Settings > Developer Options > Check the option POINTER SLOCATION

..

Swipe X1 Y1 X2 Y2 [duration(ms)]:

adb shell input swipe 100 500 100 1450 100
in this example X1=100, Y1=500, X2=100, Y2=1450, Duration = 100ms

..

LongPress X Y:

adb shell input swipe 100 500 100 500 250
we utilise the same command for a swipe to emulate a long press

in this example X=100, Y=500, Duration = 250ms

..

Event Codes Updated List:

0 -->  "KEYCODE_0" 
1 -->  "KEYCODE_SOFT_LEFT" 
2 -->  "KEYCODE_SOFT_RIGHT" 
3 -->  "KEYCODE_HOME" 
4 -->  "KEYCODE_BACK" 
5 -->  "KEYCODE_CALL" 
6 -->  "KEYCODE_ENDCALL" 
7 -->  "KEYCODE_0" 
8 -->  "KEYCODE_1" 
9 -->  "KEYCODE_2" 
10 -->  "KEYCODE_3" 
11 -->  "KEYCODE_4" 
12 -->  "KEYCODE_5" 
13 -->  "KEYCODE_6" 
14 -->  "KEYCODE_7" 
15 -->  "KEYCODE_8" 
16 -->  "KEYCODE_9" 
17 -->  "KEYCODE_STAR" 
18 -->  "KEYCODE_POUND" 
19 -->  "KEYCODE_DPAD_UP" 
20 -->  "KEYCODE_DPAD_DOWN" 
21 -->  "KEYCODE_DPAD_LEFT" 
22 -->  "KEYCODE_DPAD_RIGHT" 
23 -->  "KEYCODE_DPAD_CENTER" 
24 -->  "KEYCODE_VOLUME_UP" 
25 -->  "KEYCODE_VOLUME_DOWN" 
26 -->  "KEYCODE_POWER" 
27 -->  "KEYCODE_CAMERA" 
28 -->  "KEYCODE_CLEAR" 
29 -->  "KEYCODE_A" 
30 -->  "KEYCODE_B" 
31 -->  "KEYCODE_C" 
32 -->  "KEYCODE_D" 
33 -->  "KEYCODE_E" 
34 -->  "KEYCODE_F" 
35 -->  "KEYCODE_G" 
36 -->  "KEYCODE_H" 
37 -->  "KEYCODE_I" 
38 -->  "KEYCODE_J" 
39 -->  "KEYCODE_K" 
40 -->  "KEYCODE_L" 
41 -->  "KEYCODE_M" 
42 -->  "KEYCODE_N" 
43 -->  "KEYCODE_O" 
44 -->  "KEYCODE_P" 
45 -->  "KEYCODE_Q" 
46 -->  "KEYCODE_R" 
47 -->  "KEYCODE_S" 
48 -->  "KEYCODE_T" 
49 -->  "KEYCODE_U" 
50 -->  "KEYCODE_V" 
51 -->  "KEYCODE_W" 
52 -->  "KEYCODE_X" 
53 -->  "KEYCODE_Y" 
54 -->  "KEYCODE_Z" 
55 -->  "KEYCODE_COMMA" 
56 -->  "KEYCODE_PERIOD" 
57 -->  "KEYCODE_ALT_LEFT" 
58 -->  "KEYCODE_ALT_RIGHT" 
59 -->  "KEYCODE_SHIFT_LEFT" 
60 -->  "KEYCODE_SHIFT_RIGHT" 
61 -->  "KEYCODE_TAB" 
62 -->  "KEYCODE_SPACE" 
63 -->  "KEYCODE_SYM" 
64 -->  "KEYCODE_EXPLORER" 
65 -->  "KEYCODE_ENVELOPE" 
66 -->  "KEYCODE_ENTER" 
67 -->  "KEYCODE_DEL" 
68 -->  "KEYCODE_GRAVE" 
69 -->  "KEYCODE_MINUS" 
70 -->  "KEYCODE_EQUALS" 
71 -->  "KEYCODE_LEFT_BRACKET" 
72 -->  "KEYCODE_RIGHT_BRACKET" 
73 -->  "KEYCODE_BACKSLASH" 
74 -->  "KEYCODE_SEMICOLON" 
75 -->  "KEYCODE_APOSTROPHE" 
76 -->  "KEYCODE_SLASH" 
77 -->  "KEYCODE_AT" 
78 -->  "KEYCODE_NUM" 
79 -->  "KEYCODE_HEADSETHOOK" 
80 -->  "KEYCODE_FOCUS" 
81 -->  "KEYCODE_PLUS" 
82 -->  "KEYCODE_MENU" 
83 -->  "KEYCODE_NOTIFICATION" 
84 -->  "KEYCODE_SEARCH" 
85 -->  "KEYCODE_MEDIA_PLAY_PAUSE"
86 -->  "KEYCODE_MEDIA_STOP"
87 -->  "KEYCODE_MEDIA_NEXT"
88 -->  "KEYCODE_MEDIA_PREVIOUS"
89 -->  "KEYCODE_MEDIA_REWIND"
90 -->  "KEYCODE_MEDIA_FAST_FORWARD"
91 -->  "KEYCODE_MUTE"
92 -->  "KEYCODE_PAGE_UP"
93 -->  "KEYCODE_PAGE_DOWN"
94 -->  "KEYCODE_PICTSYMBOLS"
...
122 -->  "KEYCODE_MOVE_HOME"
123 -->  "KEYCODE_MOVE_END"
The complete list of commands can be found on: http://developer.android.com/reference/android/view/KeyEvent.html

**Comment by AlbaHoo** [comment]
rails delete_all just delete current table, destroy_all will delete the current table + all associated tables.

**Comment by AlbaHoo** [comment]
To use multiple accounts in aws-cli

$ aws configure --profile account1
$ aws configure --profile account2

$ aws dynamodb list-tables --profile account1
$ aws s3 ls --profile account2
Note:

If you name the profile to be default it will become default profile i.e. when no --profile param in the command.

**Comment by AlbaHoo** [comment]
Shell expansion:

${parameter/pattern/string}

If pattern begins with ‘/’, all matches of pattern are replaced with string
If pattern begins with ‘#’, it must match at the beginning
If pattern begins with ‘%’, it must match at the end
If string is null, delete match(es)

example: 
$ echo "${x/ab/_}" 
_abab
$ echo "${x//ab/_}" 
_abab

**Comment by AlbaHoo** [comment]
#!/bin/bash
var1="A B  C   D"
echo $var1   # A B C D
echo "$var1" # A B  C   D

**Comment by AlbaHoo** [comment]
Caused by: java.lang.UnsatisfiedLinkError: could find DSO to load: libreactnativejni.so             
	 at com.facebook.soloader.SoLoader.loadLibraryBySoName(SoLoader.java:213)
————————————————
版权声明：本文为CSDN博主「lplj717」的原创文章，遵循 CC 4.0 BY-SA 版权协议，转载请附上原文出处链接及本声明。
原文链接：https://blog.csdn.net/lplj717/java/article/details/77370979


 defaultConfig {
        applicationId "com.example.lipiao.myreactnativeapp"
        minSdkVersion 16
        targetSdkVersion 25
        versionCode 1
        versionName "1.0"
        ndk{
            abiFilters "armeabi-v7a","x86"
        }
    }
    packagingOptions {
        exclude "lib/arm64-v8a/librealm-jni.so"
    }

————————————————
版权声明：本文为CSDN博主「lplj717」的原创文章，遵循 CC 4.0 BY-SA 版权协议，转载请附上原文出处链接及本声明。
原文链接：https://blog.csdn.net/lplj717/java/article/details/77370979

**Comment by AlbaHoo** [comment]
https://blog.csdn.net/liangtianmeng/article/details/82879848

**Comment by AlbaHoo** [comment]
DELETE FROM dups a USING (
      SELECT MIN(ctid) as ctid, key
        FROM dups 
        GROUP BY key HAVING COUNT(*) > 1
      ) b
      WHERE a.key = b.key 
      AND a.ctid <> b.ctid

**Comment by AlbaHoo** [comment]
React build, blank page

Something to do with React project, there are two types of Router, BrowserHistory vs HashHistory. 
 BrowserHistory will need a local server to serve the relative files, AWS s3, nginx, a node package called "http-server" can help with this.
 HashHistory doesn't need a local server, everything in one big file, "pages" are separated by a "#" in the browser url.

add homepage: "." to package.json

**Comment by AlbaHoo** [comment]
export default async function downloadFileFromUrl(url, defaultName, headers, onFinish) {
  const response = await fetch(url, { method: 'GET', headers: headers || {} });
  const contentDisposition = response.headers.get('Content-Disposition');
  const filename = contentDisposition ? contentDisposition.split('filename=')[1] : defaultName;
  if (response.ok) {
    const blob = await response.blob();
    // Create blob link to download
    const blobUrl = window.URL.createObjectURL(blob);
    const link = document.createElement('a');
    link.href = blobUrl;
    link.download = filename;
    // Force download
    link.click();
  } else {
    const message = `Error Message: ${response.statusText}`;
    handleError(message);
  }
  if (onFinish) {
    onFinish();
  }
}

**Comment by AlbaHoo** [comment]
// IE10+ : (has Blob, but not a[download] or URL)
if (navigator.msSaveBlob) {
    return navigator.msSaveBlob(blob, fileName);
}

**Comment by AlbaHoo** [comment]
#!/bin/sh
aws='http://52.77.191.142/api'
staging='http://adbpublicstg.prod.acquia-sites.com/multimedia/scf/scf.api.php?api_url=http://52.77.191.142/api'
prod='https://www.adb.org/multimedia/scf/scf.api.php?api_url=http://52.77.191.142/api'
rm -rf build
REACT_APP_API_URL=${aws} yarn build
export AWS_DEFAULT_PROFILE=coriolis
aws s3 sync build s3://adb-frontend-preview --acl public-read
rm -rf build
REACT_APP_API_URL=${staging} yarn build && zip "build-staging-${1:-'final'}.zip" -r build
rm -rf build
REACT_APP_API_URL=${prod} yarn build && zip "build-prod-${1:-'final'}.zip" -r build

**Comment by AlbaHoo** [comment]
#!/bin/bash -e

 REMOTE_BRANCHES="`mktemp`"
 LOCAL_BRANCHES="`mktemp`"
 DANGLING_BRANCHES="`mktemp`"
 git for-each-ref --format="%(refname)" refs/remotes/origin/ | \
   sed 's#^refs/remotes/origin/##' > "$REMOTE_BRANCHES"
 git for-each-ref --format="%(refname)" refs/heads/ | \
   sed 's#^refs/heads/##' > "$LOCAL_BRANCHES"
 grep -vxF -f "$REMOTE_BRANCHES" "$LOCAL_BRANCHES" | \
   sort -V > "$DANGLING_BRANCHES"
 rm -f "$REMOTE_BRANCHES" "$LOCAL_BRANCHES"

 # prune remote branches, wait till finished, and then delete local references
git remote prune origin && cat "$DANGLING_BRANCHES" | while read -r B; do git branch -D "$B"; done
rm -f "$DANGLING_BRANCHES"

# git fetch -p && for branch in `git branch -vv | grep ': gone]' | awk '{print $1}'`; do git branch -D $branch; done

**Comment by AlbaHoo** [comment]
The arithmetic expansion can be performed using the double parentheses ((...)) and $((...))

**Comment by AlbaHoo** [comment]
"\e[A": history-search-backward
"\e[B": history-search-forward

**Comment by AlbaHoo** [comment]
# GitHub域名ip地址
13.229.188.59 github.com 

dig github.com

**Comment by AlbaHoo** [comment]
# loop over folder

i=0
for filename in ./baostock/csv/day/*.csv
do
  i=$((i + 1))
  echo "dumping $filename... [$i]"
  cat $filename | psql $production -f $daily
done

**Comment by AlbaHoo** [comment]
# loop line over a file
# while IFS= read -r line; do
#   echo "fetch daily kdata for $line"
#   python daily_kdata.py --code $line --start 1989-01-01 --end 2020-09-08
# done < codes

**Comment by AlbaHoo** [comment]
for file in $(find src -type f -name *index.less*); do mv $file "${file%%index.less}style.less"; done

**Comment by AlbaHoo** [comment]
Provisional headers are shown

CORS

**Comment by AlbaHoo** [comment]
pg_dump -U Username -h DatabaseEndPoint -a -t TableToCopy SourceDatabase > dump
cat dump | psql -h DatabaseEndPoint -p portNumber -U Username -W TargetDatabase

**Comment by AlbaHoo** [comment]
select nextval(pg_get_serial_sequence('counts', 'id'));

**Comment by AlbaHoo** [comment]
white-space: nowrap;

**Comment by AlbaHoo** [comment]
天问9问，视频问，个人怎么跨地域去起诉一个大公司的不合法问题？

**Comment by AlbaHoo** [comment]
ruby-install ruby 2.5 --no-install-deps -- --with-openssl-dir=$(brew --prefix openssl) --disable-install-doc

MacOS upgarde deprecated openssl

**Comment by AlbaHoo** [comment]
for f in *; do convert "$f" -resize 300x300 "${f: : -4}-100.png";done
for mac

**Comment by AlbaHoo** [comment]
convert out.png -resize 2516x1718 -gravity center -extent 3555x2000 out2.png
convert +append 1.png 2.png out.png
convert -append 1.png 2.png out.png

**Comment by AlbaHoo** [comment]
convert 1.png -append -bordercolor "#FFFFFF" -border 0x100 out2.png
settings such as -bordercolor, need to come before the option that uses it, such as -border

**Comment by AlbaHoo** [comment]
<meta name="viewport" content="width=device-width, initial-scale=1">

Initial match device width

**Comment by AlbaHoo** [comment]
To remove ^M in vim

This worked for me

:e ++ff=dos 
The :e ++ff=dos command tells Vim to read the file again, forcing dos file format. Vim will remove CRLF and LF-only line endings, leaving only the text of each line in the buffer.

then

:set ff=unix 
and finally

:wq

**Comment by AlbaHoo** [comment]
～～狗子～～

**Comment by AlbaHoo** [comment]
10:30 到达星巴克，开始求婚系统

**Comment by AlbaHoo** [comment]
聊天APP
A和B交流
A先设定默认的密钥
A也可以设定和B聊天的密钥
B通过别的手段获取了A的密钥，输入和A的聊天窗口进行解密

**Comment by AlbaHoo** [comment]
信息邮局：
没有用户，
用户A可以发送一条加密的数据到服务器
3天以内任何人都可以通过密码获取这个数据，数据获取以后自动删除
发送这种信息1毛钱1m.条，访问免费。

**Comment by AlbaHoo** [comment]
du -sh *|sort -h

**Comment by AlbaHoo** [comment]
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
    "AzureWebJobsDashboard": "UseDevelopmentStorage=true",

    "Bvd:BaseUrl": "https://Orbis.bvdinfo.com/api",
    "Bvd:ApiToken": "22SV3f69a1285d95e81180bd2c44fd99a584",

    "Cosmos:EndpointUrl": "https://localhost:8081",
    "Cosmos:DatabaseName": "Coriolis.Functions.Companies",
    "Cosmos:Key": "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==",

    "Gleif:BaseUrl": "https://api.gleif.org",

    "GovUk:ResourceUrls:FinancialSanctions": "https://ofsistorage.blob.core.windows.net/publishlive/ConList.csv",

    "Eu:ResourceUrls:FinancialSanctions": "https://coriolisfunctions.blob.core.windows.net/sanctions/eu-consolidated-sanctions.csv",

    "Ofac:ConnectionString": "Server=tcp:adara.database.windows.net,1433;Initial Catalog=Ofac;Persist Security Info=False;User ID=atlas;Password=izGCmDiCFev7;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;",

    "Hmrc:BaseUrl": "https://api.service.hmrc.gov.uk",
    "Hmrc:SandboxBaseUrl": "https://test-api.service.hmrc.gov.uk",
    "Hmrc:ClientId": "QHBeit7zeVRHY1cUBoGQYKhEmILt",

    "VatApi:ApiToken": "lB04SgP3bF4MVbZpEhCn869VFzEwp6Wm9kuUejWq",
    "VatApi:BaseEuUrl": "https://eu.vatapi.com/v2",
    "VatApi:BaseGlobalUrl": "https://global.vatapi.com/v2"
  }
}

**Comment by AlbaHoo** [comment]
DATEADD(year, 1, '2017/08/25')

**Comment by AlbaHoo** [comment]
统计数据分析网站
1. 积分系统
2. 上传
3. 搜索数据
例子： 百度文库。

**Comment by AlbaHoo** [comment]
cat urls_tags.csv | python -c 'import csv, json, sys; print(json.dumps([dict(r) for r in csv.DictReader(sys.stdin)]))' | jq

**Comment by AlbaHoo** [comment]
00775-15842
81bc2-6f04a
9f73f-a731e
3031c-401ba
85c88-04c5a
81ea4-b7a70
1aab3-d23bc
7e3ae-20a99
b3e62-032cd
27112-c4060
4cce6-7be8a
37fa4-c5691
01ea7-07e40
302f8-3fc6d
970b2-fafc8
f91e8-2b906

**Comment by AlbaHoo** [comment]
4DeY99nkBP1eIRv1tIRDUE8c
9KxEf4TERc72Zry8zuz9DYVi

**Comment by AlbaHoo** [comment]
docker exec -it SQLEXPRESS01 "bash"
/opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P "[password]"

USE master ;  
GO  
DROP DATABASE TestDB;  
GO
create database TestDB;
GO

**Comment by AlbaHoo** [comment]
动作捕捉的原理

**Comment by AlbaHoo** [comment]
Option3: mastercard

Doc: swagger-ish 

Desc:The Carbon Calculator API enables Issuers to provide their consumers with visibility into the environmental impact based on their spending habits.

Pro: 

Simple integration, it estimates the emission based on historical transactions.Con: 

May have some privacy issue, and this API fit more to individuals instead of a SME company.

Price is unknown, but seems free need to create a project to find out.

Option4: myclimate

Doc: swagger 

Desc:It provides dedicated APIs to calculate carbon emission, include

flights

car trips

cruises

companies

personal footprint

Pro: 

Straightforward to integrate, and the companies API looks like the right fit.

Con:

Cannot try out, need to contact info@myclimate.org for an account.

Don't know how accurate the calculation would be.

Option5: climatiq

Doc:

Price Doc:  

Desc: The api take parameters to calculate the final carbon emission.The quick guide of parameters.

Pro: 

Looks professional, rich documentation.

Free plan: <10 API requests/sec; <100,000 API calls/month.

Flexible since it has many different dimenions of params.

Con: don’t know yet.

Climatiq looks like the most professional across all the options here, personally I will choose this one as a start.

**Comment by AlbaHoo** [comment]
DefaultEndpointsProtocol=https;AccountName=coriolisotfpapidev;AccountKey=Ww3uyMlzSsFEMFiCqzX+c174jo2TU2cUsLURSCxn+ymSiEpySsKASSWvbRtDGN+4Dv2RzlGXcrJQaUIqkiYNgQ==;EndpointSuffix=core.windows.net

**Comment by AlbaHoo** [comment]
8230  cd ~/Downloads/Apache_.coriolis.systems
 8231  cat 43aad6c00663cfc1.crt gd_bundle-g2-g1.crt >> ssl-bundle.crt
 8232  cp generated-private-key.txt private.pem
 8234  mkdir nginx
 8235  mv private.pem nginx
 8236  mv ssl-bundle.crt nginx
 8237  cd nginx
 8238  scp * remoteServer:~/

644 for private.pem
777 for ssl-bundle.crt

**Comment by AlbaHoo** [comment]
file -b --mime-encoding source.file

**Comment by AlbaHoo** [comment]
101M	Shared
128G	Alba

~/Library/
995M	Nemu
1.3G	Mail
7.5G	Caches
8.3G	Application Support
8.3G	Developer
 17G	Android
 26G	Containers

~/Library/Containers
304M	com.tencent.meeting
458M	com.xiami.client
4.4G	com.tencent.xinWeChat
 20G	com.docker.docker

**Comment by AlbaHoo** [comment]
# HTML

# 过一遍 HTML/CSS/Javascript

每一个差不多30分钟
[HTML教程](https://www.w3school.com.cn/html/index.asp)
[css教程](https://www.w3school.com.cn/css/index.asp)
[js教程](https://www.w3school.com.cn/js/index.asp)


# 通过 HTML/CSS/Javascript在线测试
[HTML测试](https://www.w3school.com.cn/html/html_quiz.asp)
[CSS测试](https://www.w3school.com.cn/css/css_quiz.asp)
[JS测试 只有英文的](https://www.w3schools.com/js/js_quiz.asp)

**Comment by AlbaHoo** [comment]
小程序：
1. 注册微信开放公众平台，授权微信登陆。
1. 代理商A登陆，报备一家酒店，提供营业执照名词，税号，注册地址，备注报备酒店。
1. 30天以后自动过期。
1. 代理商B可以登陆报备同一家酒店，也需要提供营业执照名词，税号，注册地址，但是必须排队，等A过去。
1. 市场部可以登陆去审核条件，变更代理商A的报备状态（授权，驳回）
1. 授权以后其他代理商的报备自动驳回。
1. 驳回以后，代理商B的报备马上生效。

**Comment by wenhaohoo** [comment]
hi

**Comment by AlbaHoo** [comment]
# Visual studio: preferences -> location

## x86

/usr/local/share/dotnet/x64/dotnet

## arm64

/usr/local/share/dotnet/dotnet

# .zshrc

## x86

export PATH="/usr/local/share/dotnet/x64:$PATH"

## arm64

export PATH="/usr/local/share/dotnet:$PATH"

**Comment by AlbaHoo** [comment]
微信小程序一家人玩扑克牌

**Comment by AlbaHoo** [comment]
dd if=/dev/rdisk4 of=sd_backup.dmg status=progress bs=16M

https://blog.jaimyn.dev/the-fastest-way-to-clone-sd-card-macos/

From that output we can see that our SD card must be /dev/disk4 as our card is 32GB in size and has a fat32 and linux partition (standard for most raspberry pi images). You should add an r in front of disk4 so it looks like this /dev/rdisk4. The r means when we’re copying, it will use the “raw” disk. For an operation like this, it is much more efficient.

Copy the disk image (dmg) to your SD card
You’ll first need to unmount your SD card. Do not click the eject button in finder, but run this command, replacing 4 with whatever number you identified as your sd card sudo diskutil unmountDisk /dev/disk4.

Then to copy the image, run the following command:

sudo gdd of=/dev/rdisk4 if=sd_backup.dmg status=progress bs=16M

Tip: you can experiment with different numbers for the block size by replacing bs=16M with larger or smaller numbers to see if it makes a difference to the speed. I’ve found 16M the best for my hardware.

**Comment by AlbaHoo** [comment]
watermark remove

**Comment by AlbaHoo** [comment]
text image ocr

**Comment by AlbaHoo** [comment]
sudo spctl --master-disable

**Comment by AlbaHoo** [comment]
<video src="blob:https://www.example.tv/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx">
  <source src="https://cdn.example.tv/api/media/tv/xyzxyz/index" type="application/x-mpegurl">
</video>
I also had blob: URL in video/@src, but by watching Developer tools > Network during playback it turned out that video/source/@src was URL for m3u8 playlist.

An m3u8-backed video can be readily downloaded by either:

ffplay -i "https://cdn.example.tv/api/media/tv/xyzxyz/1080/index.m3u8"
ffmpeg -i "https://cdn.example.tv/api/media/tv/xyzxyz/1080/index.m3u8" -codec copy file.mkv
tl;dr - blob URL sounds like the binary you want to get but there might be easier way to get the video. Just check out Network tab in Dev tools while you play the video to see what you are actually fetching.

**Comment by AlbaHoo** [comment]
defaults write com.microsoft.Word AppleLanguages '("zh_CN")'
defaults write com.microsoft.Excel AppleLanguages '("zh_CN")'
defaults write com.microsoft.Powerpoint AppleLanguages '("zh_CN")'


defaults write com.microsoft.Word AppleLanguages '("en")'
defaults write com.microsoft.Excel AppleLanguages '("en")'
defaults write com.microsoft.Powerpoint AppleLanguages '("en")'

**Comment by AlbaHoo** [comment]
qczf-mwrm-jshe-uynp-hzsf stripe

**Comment by AlbaHoo** [comment]
create a scripe to delete old files: delete_old_log

```
#!/bin/sh

# delete all log files whole last modified time was over 6 days ago
file=$(find /srv/backend/helong/log/frame.log.* -mtime +$1)

if [ -z "$file" ]
then
  echo "No log fine found"
else
  echo "delete these files: $file"
  rm $file
fi
```

chmox +x delete_old_log


Then create cronjob to run 

 crontab -e

0 23 * * * /home/rails/script/delete_old_log 7

**Comment by AlbaHoo** [comment]
make a tool page:

1. formatted json from json
2. format string to array,json in one line or multiple line

**Comment by AlbaHoo** [comment]
WireGuard:

docker run -d \
  --name=wg-easy \
  -e LANG=de \
  -e WG_HOST=🚨YOUR_SERVER_IP \
  -e PASSWORD=🚨YOUR_ADMIN_PASSWORD \
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
  ghcr.io/wg-easy/wg-easy

Ubuntu 18: apt-get install wireguard-dkms

Dashboard: "YOUR_SERVER_IP:51820" login with YOUR_ADMIN_PASSWORD

Add a client, download the file.

Rename the downloaded file to have .conf


default.conf preview
```
[Interface]
PrivateKey = 
Address = 
DNS = 1.1.1.1

[Peer]
PublicKey = 
PresharedKey = 
AllowedIPs = 0.0.0.0/0, ::/0
PersistentKeepalive = 0
Endpoint = your_ip:51820

```

Download WireGuard profile and import tunnel

**Comment by AlbaHoo** [comment]
1827  sudo apt-get update -y
 1828  sudo apt-get install php7.3-fpm -y
 1829  exit
 1830  ls
 1831  vim  /etc/nginx/sites-available/cms
 1832  sudo ln -s /etc/nginx/sites-available/cms /etc/nginx/sites-enabled/cms
 1833  ls /etc/nginx/sites-enabled
 1834  nginx -t
 1835  systemctl reload nginx
 1836  mysql
 1837  sudo mysql
 1838  sudo systemctl start mysqld.service
 1839  sudo systemctl start mysql.service
 1840  locale mysql
 1841  locale mysql.service
 1842  which mysql
 1843  exit
 1844  ls
 1845  cp -r PbootCMS/* /var/www/html/
 1846  chmod -R a+w /var/www/html
 1847  ls /var/www/html/static/backup/sql/
 1848  mysql -uroot -pHUdqh pbootcms
 1849  vim /var/www/html/config/database.php
 1850  systemctl reload nginx
 1851  ls
 1852  cd /etc/php/7.2/
 1853  ls
 1854  cd fm
 1855  cd fpm/
 1856  ls
 1857  vim php.ini
 1858  history
 1859  systemctl list
 1860  systemctl -l
 1861  systemctl -l | grep php
 1862  systemctl php7.2-fpm.service restart
 1863  systemctl restart php7.2-fpm.service
 1864  vim php.ini
 1865  systemctl restart php7.2-fpm.service
 1866  vim php.ini
 1867  apt-get install php-gd
 1868  systemctl restart php7.2-fpm.service
 1869  apt-get install php-mbstring
 1870  apt-get install php-curl
 1871  mysql -uroot -pHUdqh
 1872  history
 1873  exit
 1874  /etc/mysql/mysql.conf.d/mysqld.cnf
 1875  vim /etc/mysql/mysql.conf.d/mysqld.cnf
 1876  exit
 1877  mysql
 1878  exit
 1879  mysql
 1880  ls
 1881  history
 1882  vim /var/www/html/config/database.php
 1883  exit
 1884  Reading package lists... Done
 1885  sudo add-apt-repository universe
 1886  sudo apt install php-fpm php-mysql
 1887  ls
 1888  vim /etc/nginx/sites-available/
 1889  cp /etc/nginx/sites-available/tenty /etc/nginx/sites-available/cms
 1890  sudo -E bash
 1891  sudo apt install mysql-server
 1892  sudo mysql_secure_installation
 1893  sudo mysql
 1894  sudo mysql -uroot -p
 1895  ls
 1896  cd ~ && git clone https://gitee.com/hnaoyun/PbootCMS.git
 1897  cp -r PbootCMS/* /var/www/html/
 1898  sudo -E bash
 1899  mysql -uroot -pHUdqh
 1900  service mysql restart
 1901  sudo service mysql restart
 1902  mysql -uroot -pHUdqh
 1903  cat /etc/mysql/mysql.conf.d/mysqld.cnf
 1904  sudo -E bash
 1905  mysql -umysql -pHUdqh
 1906  mysql -uroot -pHUdqh
 1907  sudo -E bash
 1908  sudo mysql
 1909  mysql
 1910  mysql -uroot -pHUdqh
 1911  sudo mysql
 1912  cd /var/www/html/
 1913  ls
 1914  sudo mysql
 1915  mysql
 1916  mysql -uroot -pHUdqh
 1917  sudo service mysql restart
 1918  mysql -uroot -pHUdqh
 1919  mysql
 1920  sudo mysql
 1921  sudo -E bash
 1922  history
 1923  sudo service mysql restart
 1924  history | grep restart
 1925  history

**Comment by AlbaHoo** [comment]
00775-15842 
81bc2-6f04a 
9f73f-a731e 
3031c-401ba 
85c88-04c5a 
81ea4-b7a70 
1aab3-d23bc 
7e3ae-20a99 
b3e62-032cd 
27112-c4060 
4cce6-7be8a 
37fa4-c5691 
01ea7-07e40 
302f8-3fc6d 
970b2-fafc8 
f91e8-2b906

**Comment by AlbaHoo** [comment]
cat nace-es-dedup.csv | python -c 'import csv, json, sys; print(json.dumps([dict(r) for r in csv.DictReader(sys.stdin)]))' | jq 'map({(.Code): .ES}) | add'

**Comment by AlbaHoo** [comment]
Stripe: ptdx-nlgh-wflt-zqhg-jybg

**Comment by AlbaHoo** [comment]
https://play.google.com/apps/internaltest/4701544591654641431

**Comment by AlbaHoo** [comment]
23wBBceo66Ygu009

**Comment by AlbaHoo** [comment]
overlayStyle={{ position: 'fixed' }}

**Comment by AlbaHoo** [comment]
docker builder prune -f

**Comment by AlbaHoo** [comment]
test
